packages/react/src/hooks/useSelectedDevice.ts

```diff
@@ -0,0 +1,72 @@
+import { type LocalAudioTrack, type LocalVideoTrack } from 'livekit-client';
+import * as React from 'react';
+import { useMediaDevices } from './useMediaDevices';
+
+/**
+ * /**
+ * The `useSelectedDevice` hook returns the current selected device (audio or video) of the participant.
+ *
+ * @example
+ * ```tsx
+ * const { selectedDevice } = useSelectedDevice({
+ *   kind: 'videoinput',
+ *   track: track,
+ * });
+ *
+ * <div>
+ *  {selectedDevice?.label}
+ * </div>
+ * ```
+ * @public
+ */
+export function useSelectedDevice<T extends LocalVideoTrack | LocalAudioTrack>({
```

Comment:
within the livekit-client SDK there's the concept of an active device which is currently used by the user. 
That's not as useful for prejoin as of now, but it would mirror the implementation in the client SDK more closely.